pkg/proxy/service.go

```diff
@@ -322,13 +320,10 @@ func (sct *ServiceChangeTracker) serviceToServiceMap(service *v1.Service) Servic
 		return nil
 	}
 
-	if len(service.Spec.ClusterIP) != 0 {
-		// Filter out the incorrect IP version case.
-		// If ClusterIP on service has incorrect IP version, service itself will be ignored.
-		if sct.isIPv6Mode != nil && utilnet.IsIPv6String(service.Spec.ClusterIP) != *sct.isIPv6Mode {
-			utilproxy.LogAndEmitIncorrectIPVersionEvent(sct.recorder, "clusterIP", service.Spec.ClusterIP, service.Namespace, service.Name, service.UID)
-			return nil
-		}
+	clusterIP := utilproxy.GetClusterIPByFamily(sct.ipFamily, service)
+	if clusterIP == "" {
+		utilproxy.LogAndEmitIncorrectIPVersionEvent(sct.recorder, "clusterIPs", strings.Join(service.Spec.ClusterIPs, ","), service.Namespace, service.Name, service.UID)
```

Comment:
previous comment still applies; this should not be logged as an error any more. (You fixed this in `newBaseServiceInfo` but not here.)

Reply:
Those are fundamentally two different situations. `newBaseServiceInfo` deals with filtering external LB IPs and source ranges. Having an unexpected family there became an expected condition because a dual stack service will have just that. I have changed the log level in `newBaseServiceInfo` to reflect we are `ignoring xxx, but we are cool..no error here ` situation

the above asks the changeTracker which operates on a single family - set during initialization - to map service that does not really support `changeTracker.family` (because it does not have a ClusterIP of the same family)This is an error case. It shouldn't happen really. But incase it happens an error log will help whoever caused the situation to ratify it.